firedrake/solving_utils.py

```diff
@@ -21,6 +23,101 @@ def _make_reasons(reasons):
 SNESReasons = _make_reasons(PETSc.SNES.ConvergedReason())
 
 
+if get_config()["options"]["petsc_int_type"] == "int32":
+    DEFAULT_KSP_PARAMETERS = {"mat_type": "aij",
+                              "ksp_type": "preonly",
+                              "ksp_rtol": 1e-7,
+                              "pc_type": "lu",
+                              "pc_factor_mat_solver_type": "mumps",
+                              "mat_mumps_icntl_14": 200}
+else:
+    DEFAULT_KSP_PARAMETERS = {"mat_type": "aij",
+                              "ksp_type": "preonly",
+                              "ksp_rtol": 1e-7,
+                              "pc_type": "lu",
+                              "pc_factor_mat_solver_type": "superlu_dist"}
+
```

Comment:
I remember we discussed this in the meeting, but I forget. Why do we use `superlu` here instead of `mumps`?

Reply:
mumps doesn't have a 64bit mode.